wasmbinding/message_plugin.go

```diff
@@ -201,6 +201,20 @@ func (m *CustomMessenger) PerformSubmitTx(ctx sdk.Context, contractAddr sdk.AccA
 		InterchainAccountId: submitTx.InterchainAccountId,
 		Timeout:             submitTx.Timeout,
 	}
+	params := m.Keeper.GetParams(ctx)
+	if uint64(len(submitTx.Msgs)) > params.GetMsgSubmitTxMaxMessages() {
```

Comment:
you validate only the wasm bindings. But user(contract) still can craft the proto(stargate) msg and ignore the limitation. am i right?

Reply:
I have no idea.

CC @pr0n00gler @oopcode

Reply:
Moved validation from `PerformSubmitTx()` to keeper's `SubmitTx()`